lib/audit_log/audit_log/middleware.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+# Common space for Insights API stuff
+module Insights
+  module API
+    module Common
+      class AuditLog
+        # Request events listener, capturing for evidence
+        class Middleware
```

Comment:
This represents the top level object, the middleware itself. If I understand correctly, it's intended to use this like:

```rb
config.middleware.use Insights::API::Common::AuditLog::Middleware
```

If that's the case, we need to ensure all the `require`s start from here so things like `AuditLog.logger` are already defined properly.

Reply:
Yes, the usage is correct.

The logger is setup with `Insights::API::Common::AuditLog.setup(Logger.new)`. Require has no effect on it, only the first use of the logger. The default one can be overridden by the mentioned setup.